### PR TITLE
Kill NEEDS_LOC and move cmdline descriptions into resources

### LINKED ISSUE #4155: Audit the code for all `NEEDS_LOC(x)` and fix them
This macro should be turned into a build break at our earliest convenience. If we ship with this macro in our code, we have failed.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
Fixes #4155.

## Validation steps

```
Summary: Total=23, Passed=22, Failed=1, Blocked=0, Not Run=0, Skipped=0
```

The failing test is the same one as before. It is not germane to this pull request.

### CODE REVIEW

File: src/cascadia/TerminalApp/AppCommandlineArgs.cpp [hunk 1]
```diff
@@ -163,7 +164,7 @@ void AppCommandlineArgs::_buildParser()
 // - <none>
 void AppCommandlineArgs::_buildNewTabParser()
 {
-    _newTabCommand.subcommand = _app.add_subcommand("new-tab", NEEDS_LOC("Create a new tab"));
+    _newTabCommand.subcommand = _app.add_subcommand("new-tab", winrt::to_string(RS_(L"CmdNewTabDesc")));
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/AppCommandlineArgs.cpp]: does this seem like an ask for a new macro, `RS_A`?

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/AppCommandlineArgs.cpp]: _it kinda does_

Comment (miniksa) [reply src/cascadia/TerminalApp/AppCommandlineArgs.cpp]: Works for me.

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/AppCommandlineArgs.cpp]: nm had an even crazier and better idea

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/AppCommandlineArgs.cpp]: nm the idea was great but the center could not hold

**Comment by DHowett-MSFT** [comment]
I'm planning to follow up with a separate TermControl change. This one is specific for NEEDS_LOC.

**Pull request merged**